### DISCUSSION TITLE
Looking for samples for multicore SMP

### DISCUSSION THREAD

**Discussion by hcye2022** [zephyrproject-rtos/zephyr#77131] [General]
From the docs it appears that multicore SMP has been tested on esp32 and qemu_x86. But I search the codebase and did not find any sample using the SMP API (e.g. k_thread_cpu_mask_xxx). Can someone point me to such code, if they exist?

**Comment by tbursztyka** [comment]
A grep shows some relevant tests : 
- tests/arch/common/timing
- tests/boards/intel_adsp/smoke

 See tests in tests/kernel/smp*  there are way more relevant code (using k_thread_cpu_pin for instance)

**Reply by hcye2022** [reply]
Thanks for the info. Unfortunately that is for unit test only, not executable on a real board. And it has only k_thread_cpu_pin(), so it is AMP not SMP.

**Comment by rftafas** [comment]
While it is true that SMP has been tested in ESP32, it is also true that it is not working properly. PTAL: https://github.com/zephyrproject-rtos/zephyr/issues/56011
